### ISSUE TITLE
Bluetooth volume extremely high

### ISSUE THREAD

**Issue by github-account1111** [UnigramDev/Unigram#2685]
**Describe the bug**
Just blasted my ears when trying to talk to someone through a Bluetooth headset.

**To Reproduce**
1. Connect a Bluetooth headset.
1. Call someone.

**Expected behavior**
Call volume respects the system setting.

**Version Info**
 - Unigram: 8.6
 - Windows: 10 1809

**Additional context**

**Comment by FrayxRulez** [comment]
Did you try to decrease the volume level?

**Comment by FrayxRulez** [comment]
Operating systems usually have two different audio channels for media and communication, with independent volume levels. Unigram uses the communication channel while you're in a call.

**Comment by thirtndev** [comment]
@github-account1111 it's not a bug. it's a feature. we can control both of our audio channels for media and communication.

**Comment by github-account1111** [comment]
> Did you try to decrease the volume level?

Yes ofc.
It wasn't decreasing.

**Comment by github-account1111** [comment]
> @github-account1111 it's not a bug. it's a feature. we can control both of our audio channels for media and communication.

Not sure what you're saying.
It shouldn't disrespect OS settings.
That's bad practice and can even lead to things like hearing loss.

**Comment by FrayxRulez** [comment]
It does in fact respect system settings by using the dedicated communication audio of your computer.
Just to state it clearly: there's no way (technically, the code to do this doesn't exist) to not-respect your system settings.

**Comment by MarcAnt01** [comment]
@github-account1111 to be clear, it has different calibration rather than the standard audio for music etc. , but that's not something coming from Unigram that just picks the proper audio channel for the calls (where audio is usually a bit louder compared to music). This is the same thing when you are on your phone and making a call the minimum audio is higher compared to minimum multimedia audio

**Comment by FrayxRulez** [comment]
Unigram and Telegram Desktop are two different softwares, and they behave in different ways. Telegram Desktop doesn't use communication optimized audio channels (@john-preston is there any reason for this btw?) and this is why you don't have the same behavior there.
You can try to have a call using any other communication software (Skype, Slack, Google Meets to mention some) and you will notice that they all behave the same way as Unigram.

Also, I'd ask you to be a bit more respectful here, we're all trying to explain you how this stuff works but you just pretend to be right without even trying to understand what we're telling you.

**Comment by github-account1111** [comment]
I use Signal, and it definitely doesn't suffer from this.
For one, both the desktop Signal client and Telegram Desktop let you set which mic and speaker device to use, unlike Unigram.
And yes, it does let you use hands-free telephony; not sure why you think otherwise:
![image](https://user-images.githubusercontent.com/5728319/159377673-39debcbc-a04a-4eac-b67e-4d3d66cf2513.png)

>Unigram and Telegram Desktop are two different softwares

You're missing the point.
Me pointing out that the official app doesn't have this issue was in response to this statement;

>Just to state it clearly: there's no way (technically, the code to do this doesn't exist) to not-respect your system settings.

In other words, me point it out was to show that there is obviously a way, because otherwise the official client would suffer from this too.

**Comment by MarcAnt01** [comment]
I think we have already pointed out the reasons, I’n sorry for your issues with too loud audio, I had it as well on one Android phone and ended up manually tweaking the call channel, I don’t know if this is possible on Windows as well. Whatever as Fela said many times, Unigram just uses the default audio channel, if it’s faulty it’s not definitely Unigram’s fault

**Comment by github-account1111** [comment]
>Also, I'd ask you to be a bit more respectful here, we're all trying to explain you how this stuff works but you just pretend to be right without even trying to understand what we're telling you.

Just giving you the taste of you own medicine.
I suggest you follow your own advice and try and listen to the user first instead of having a condescending holier than thou stance and immediately assuming he's wrong and "it's not a bug but a feature."
I wouldn't have even opened the issue had it not blasted in my ears at 100% volume leaving me stuttered with no ability to turn it down.

**Comment by github-account1111** [comment]
Lol nice development model.
Just assume every bug is a feature, ignore what the user says and just close every issue.

**Comment by github-account1111** [comment]
>if it’s faulty it’s not definitely Unigram’s fault

It's not faulty, and I've already pointed it out by giving examples of other apps that use it properly (Signal).
Never come across a dev that simply refuses to read or take any responsibility for their project 👍

**Comment by FrayxRulez** [comment]
> Not sure you know what "in fact" means.

I mean that most softwares (especially UWP ones that run inside a sandbox) can't take over your computer audio.
I have no access to APIs to change your system volume settings (they're not allowed by Microsoft to be used in the Microsoft Store and they're detected by a certification process, so the app won't be approved for distribution there) and there's no concept on Windows of "playing audio on its own": to reproduce audio you have to open a _software_ audio device (and you as a developer have no direct access to the hardware) and stream your data to it. It's the software device to then process the data and play it through your hardware, and I have absolutely no control over this process.
Unigram has calls since 2016 and it uses a standard library developed by Google and optimized for Windows by Microsoft itself since 2019. You're the first one in **6 years** reporting this issue, and I can guarantee you're not the first one using calls in this software.
I of course read carefully user reports before commenting them, and in this case I'm 100% sure that the problem relies on your machine configuration.
You managed to turn this place into a circus, and I don't think that anyone would have closed this issue otherwise.

**Comment by FrayxRulez** [comment]
By the way, you managed to pick probably the only software apart from Telegram Desktop not using specific communication channel for calls 😂

**Comment by makisukurisu** [comment]
> blasted in my ears at 100% volume leaving me stuttered with no ability to turn it down.

**Comment by gus33000** [comment]
![image](https://user-images.githubusercontent.com/3755345/159433691-da735dba-20df-4051-9aa6-f80aa3bdc9fa.png)

Some Windows apps make use of Windows native VoIP platform APIs. As a result if an audio device is detected as "headset" it will use this over other sources. I've attached above a picture of a list of audio devices when using bluetooth earphones. You can see there's "headphones" and there's also "headset".

When you call using Unigram the audio source used will be "Headset". Unfortunately for most of us Windows does not let you adjust the volume of it on the fly, only the main source, and it defaults to 100% volume, as shown below:

![image](https://user-images.githubusercontent.com/3755345/159433984-a1ab0c8b-851c-49f4-a2af-b7d6249e0c9c.png)

I would suggest trying to adjust these settings under Settings -> System -> Audio -> All sound devices, or if you're using an old Windows 10, find it under Control Panel.

**Comment by gus33000** [comment]
![image](https://user-images.githubusercontent.com/3755345/159434519-6e4b3eb3-63fd-4691-ae1c-6a64d1150f5d.png)

On the subject of classic control panel you can also find the setting for reducing or not other volume under it. (See above)

**Comment by github-account1111** [comment]
> You managed to turn this place into a circus, and I don't think that anyone would have closed this issue otherwise.

Ah yes, more refusing to take any responsibility and just blaming others.

Just for you personally, I just installed Skype, and it's precisely identical to Telegram Desktop:
![image](https://user-images.githubusercontent.com/5728319/159587936-6a48e067-3f57-4f38-ab8e-26ffd0ea10b6.png)
I made a test call to make sure and had absolutely zero problems, just like on Tg Desktop.

So what do we have?
You claiming Tg Desktop doesn't support something, when it turns out it does.
You then claim Tg Desktop did it wrong, that's why it's working there (lol).
I then explain how other apps, namely Signal, also behave properly - your response is Signal is "the only other badly written app how dare you pick that specific one" (there's probably like 5 or 6 major messengers total that people use for calls so like, what?) and say look at Skype instead.
I look at Skype, and it _behaves the exact same as both Signal and Tg Desktop_.

"But no-no, it's a feature.
I'll just continue with the condescending tone and accuse the user of creating a circus even though it _I_ who got all defensive with it's not a bug it's a feature bs."

**Comment by FrayxRulez** [comment]
There are literally other 4 people here telling you the same thing, I'm tired.
I will just ban you from the repository (I guess that this is what you were trying to archive by using an anonymous account?)

**Comment by github-account1111** [comment]
>P.S. Call mode volume is actually about 80-90%, not 100, but I doubt that can help with hurting ears.

This is the first actually useful comment, thank you for that.
This:
>you will be able to change volume
is not the case.
Been trying to get it to the main dev since the beginning.

**Comment by FrayxRulez** [comment]
Sorry but I'm not your internal IT.

**Comment by thirtndev** [comment]
@github-account1111 you can use other apps if you have problem with unigram.
you didn't pay for this so you can't tell how they should do this. be respectful.
